File: Model/ToDoList.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace ReminderApi.Model
{
    public class ToDoList : BaseEntity
    {

        public ToDoList()
        {

        }

        public ToDoList(string name)
        {
            Name = name;
        }

        public int Id { get; set; }
        public string Name { get; set; }

        public ICollection<Task> Tasks { get; set; }
    }
}

File: src/app/Models/todolist.ts
import { Task } from './task';

export class ToDoList {
  Id: number;
  Name: string;
  Tasks: Task[];
  DateCreated: Date;
  DateModified: Date;

  public populate(res: any) {
    this.Id = res.id;
    this.Name = res.name;
    this.Tasks = [];
    if (res.tasks != null) {
      res.tasks.forEach(element => {
        this.Tasks.push(new Task().populate(element, this));
      });
    }
    this.DateCreated = res.dateCreated;
    this.DateModified = res.dateModified;
    return this;
  }
}
